Upgrade vue-eslint-parser (#1849)
vuejs/eslint-plugin-vue: docs/user-guide/README.md

#### The variables used in the `<template>` are warned by `no-unused-vars` rule

You need to use [vue-eslint-parser] v9.0.0 or later.

> Previously you had to use the [vue/script-setup-uses-vars](../rules/script-setup-uses-vars.md) rule, but now you don't.

Reviewer: Nicer wording:

```suggestion
Previously you had to use the [vue/script-setup-uses-vars](../rules/script-setup-uses-vars.md) rule, this is no longer needed.
```

Author: Thank you so much!

#### Compiler macros such as `defineProps` and `defineEmits` generate `no-undef` warnings

You need to use [vue-eslint-parser] v9.0.0 or later.

> Previously you had to use the `vue/setup-compiler-macros` environment, but now you don't.

Reviewer: ```suggestion
Previously you had to use the `vue/setup-compiler-macros` environment, this is no longer needed.
```
